tests/util/pilot_server.go

```diff
@@ -115,11 +115,20 @@ func setup() error {
 	// This limits the options -
 	stop = make(chan struct{})
 
+	// When debugging a test or running locally it helps having a static port for /debug
+	// "0" is used on shared environment (it's not actually clear if such thing exists since
+	// we run the tests in isolated VMs)
+	pilotHttp := os.Getenv("PILOT_HTTP")
+	if len(pilotHttp) == 0 {
+		pilotHttp = "0"
+	}
+	pilotHttpPort, _ := strconv.Atoi(pilotHttp)
+
 	// Create a test pilot discovery service configured to watch the tempDir.
 	args := bootstrap.PilotArgs{
 		Namespace: "testing",
 		DiscoveryOptions: envoy.DiscoveryServiceOptions{
-			Port:            0, // An unused port will be chosen
+			Port:            pilotHttpPort,
```

Comment:
since you removed xds.go, can you also remove the rdsv2 cli option from pilot/cmd/pilot-discovery/main.go ?